### PR TITLE
CRM-19014: Add summery/overlay icons to contacts in activities list

### PR DESCRIPTION
**Pull request by niels-heinemann**
When listing activities, contacts in source, target and assignee columns are enriched with the contact type icons opening summary overlay on hover.
See [Improvement CRM-19014](https://issues.civicrm.org/jira/browse/CRM-19014)

### CODE REVIEW

File: CRM/Activity/BAO/Activity.php [hunk 1]
```diff
@@ -2517,7 +2517,12 @@ public static function getContactActivitySelector(&$params) {
           $activity['source_contact_name'] = $values['source_contact_name'];
```
Comment (jitendrapurohit) [inline CRM/Activity/BAO/Activity.php]: I think we also need to add `$srcTypeImage` here ?

Comment (nielosz) [inline CRM/Activity/BAO/Activity.php]: This one isn't linked because you just view this contact. So I thought there's also no need for a summary overlay.

**Comment by jitendrapurohit** [comment]
while working on the QA, the contact images are shown on the listing properly, but it looks little weird when we have multiple target/assignee contacts for the activity.

For around 4-5 assignee, the images are not with their respective contact names.

Screenshot - http://goo.gl/qoJr2i

**Comment by nielosz** [comment]
Jesses, you're right. I'll try adding a nowrap span?! Do you know any class in civicrm achieving nowrap?

**Comment by jitendrapurohit** [comment]
There's a class named `nowrap` which does that thing, but I'm not sure that would be helpful here. I think we need to form a proper link for display when updating the variable itself (`$activity['assignee_contact_name']`)

**Comment by nielosz** [comment]
Hmm, I'm not sure if I got your point. I just thought about passing the icon to `CRM_Utils_System::href()`. Do you mean this?

**Comment by nielosz** [comment]
I think, best would be to fix `CRM_Contact_BAO_Contact_Utils::formatContactIDSToLinks()` to recognize it's `$addEditLink` param and add a `$addSummaryIcon` param which puts a new element to each contacts result array.

**Comment by nielosz** [comment]
No, that wouldn't work too :/

**Comment by jitendrapurohit** [comment]
I mean variable `$activity['assignee_contact_name']` needs to modified such that each contact link gets surrounded by separate `<div>` element. Then adding a `nowrap` class should have each contact on a separate line in the cell.

Something like:

    <div class='nowrap'>
      <summary link1><contact link1>
    </div>
    <div class='nowrap'>
      <summary link2><contact link2>
    </div>

**Comment by nielosz** [comment]
Are you sure? Inside the foreach-loop we do what you suppose. I'm gonna show what I mean...

**Comment by nielosz** [comment]
nowrap seems too hard with long contact names

**Comment by jitendrapurohit** [comment]
@nielosz Checked this for single/multiple contacts and it seems to be working properly now on the Contact Summary page :-)

When I look at the Activities on the dashboard, it seems the summary doesn't display the values for the contact when hovered on the summary/overlay icon. 

Screenshot http://goo.gl/sjgF1I. Can you replicate this as well ?

**Comment by nielosz** [comment]
Hey @jitendrapurohit, in the dashboard it works but has to be styled or disabled. But I can reproduce it on the enlarged "fullscreen"-view of the dashlet. AJAx result is correct, I'll investigate.

**Comment by nielosz** [comment]
Problem is that jquery-datatable's css sets the background color of it's children to white which affects the overlay (which sets white text-color of field values to white). Do you know where I may override this? Any places in civi's css files to prefer?

**Comment by jitendrapurohit** [comment]
Yeah, there's a handling [done in civicrm.css](https://github.com/civicrm/civicrm-core/blob/master/css/civicrm.css#L2399-L2405) that overrides the jquery-datatable's css, which currently works for Contact Summary page only as it seems we don't have an id of crm-container in dashlet, due to which these css fail to apply after this PR.

Alternatively, I think we should not expose this on the dashlet, as it will only be possible to view the summary if the dashlet is maximised to full screen. So, can you please modify this PR to not affect when the `$context = 'dashlet'` ?

Thanks!

**Comment by nielosz** [comment]
Jep, I agree and will update later today

**Comment by jitendrapurohit** [comment]
@nielosz Can you please fix the style warnings suggested by Jenkins ?

**Comment by nielosz** [comment]
In my poor understanding the remaining jenkins error is independant from my PR?

**Comment by jitendrapurohit** [comment]
Yes, seems unrelated.

**Comment by jitendrapurohit** [comment]
This can be merged. I'll be creating another PR fixing some minor improvements to the changes.

**Comment by jitendrapurohit** [comment]
Additional PR created here https://github.com/civicrm/civicrm-core/pull/8649

@nielosz Can you please verify if it doesn't affect your requirements?

Thanks!

**Pull request merged**